webrtc.html

```diff
@@ -9849,6 +9851,40 @@ <h2>Methods</h2>
         </section>
       </div>
     </section>
+    <section>
+      <h3>canInsertDTMF update</h3>
+      <p>To <dfn>update the [[\CanInsertDTMF]]</dfn> slot of an <code><a>RTCDTMFSender</a></code>
+      instance <var>dtmfSender</var>, the user agent MUST queue a task that runs the following steps:</p>
+      <ol>
+        <li>Let <var>sender</var> be the <code><a>RTCRtpSender</a></code> associated with
+        <var>dtmfSender</var>.</li>
+        <li>Let <var>transceiver</var> be the <code><a>RTCRtpTransceiver</a></code>
+        associated with <var>sender</var>.</li>
+        <li>Let <var>pc</var> be the <code><a>RTCPeerConnection</a></code> associated with
+        <var>transceiver</var>.</li>
+        <li>Check whether <var>pc</var>'s <a>[[\ConnectionState]]</a> slot does not have the value
+        <code>"connected"</code> or <code>"completed"</code>. If so, set <var>dtmfSender</var>'s
+        <a>[[\CanInsertDTMF]]</a> slot to <code>false</code> and abort these steps.</li>
+        <li>Check whether <code><var>sender</var>.track</code> is unset.
+        If so, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to
+        <code>false</code> and abort these steps.</li>
+        <li>Check whether <code><var>sender</var>.track.kind</code> does not have the value
+        <code>"audio"</code>. If so, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a>
+        slot to <code>false</code> and abort these steps.</li>
+        <li>Check whether <var>transceiver</var>'s <a>[[\CurrentDirection]]</a> slot is unset.
+        If so, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+        and abort these steps.</li>
+        <li>Check whether <var>transceiver</var>'s <a>[[\CurrentDirection]]</a> slot
+        does not have the value <code>"sendrecv"</code> or <code>"sendonly"</code>. If so, set
+        <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+        and abort these steps.</li>
+        <li>Check whether for some value of <var>j</var> ranging from 0 to the number of
+        codecs, <code><var>sender</var>.getParameters().codecs[<var>j</var>].mimetype</code>
+        has the value <code>"audio/telephone-event"</code>. If not, set <var>dtmfSender</var>'s
+        <a>[[\CanInsertDTMF]]</a> slot to <code>false</code> and abort these steps.</li>
```

Comment:
I note that we have not added any check that the script could not do itself, so we're basically just adding convenience. Would it make sense to add something like
```
If there is any other reason why <code><var>sender</var></code> would not be able to insert DTMF , set <var>dtmfSender</var>'s  <a>[[\CanInsertDTMF]]</a> slot to <code>false</code> and abort these steps.
```

Reply:
The text is attempting to describe behavior exactly so that browsers can behave similarly.  So if there is a missing check, I would prefer it be described explicitly.

Reply:
I was after that there might be a reason that is not observable for the script why DTMF can't be sent. Similar to that the UA may fire an `negotiationneeded` event at any time even if the script has not done anything. But perhaps that is not needed - let's skip it.